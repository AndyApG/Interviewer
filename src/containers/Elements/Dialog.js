/* eslint-disable no-shadow */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Dialog as DialogComponent } from '../../components/Elements';
import modal from '../../behaviours/modal';

/**
  * A modal window which can be toggled open an closed.
  * @extends Component
  */
class Dialog extends Component {

  confirmModal = () => {
    this.props.close();
    this.props.onConfirm();
  };

  cancelModal = () => {
    this.props.close();
    this.props.onCancel();
  };

  render() {
    const {
      title,
      children,
      show,
      hasCancelButton,
      type,
    } = this.props;

    return (
      <DialogComponent
        show={show}
        title={title}
        hasCancelButton={hasCancelButton}
        type={type}
        onConfirm={this.confirmModal}
        onCancel={this.cancelModal}
      >
        {children}
      </DialogComponent>
    );
  }
}

Dialog.propTypes = {
  type: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  hasCancelButton: PropTypes.bool,
  show: PropTypes.bool,
  children: PropTypes.any,
  onCancel: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
  close: PropTypes.func.isRequired,
};

Dialog.defaultProps = {
  show: false,
  hasCancelButton: true,
  children: null,
};


export default modal(Dialog);
