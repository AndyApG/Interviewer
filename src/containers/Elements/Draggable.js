import React, { Component } from 'react';
import { connect } from 'react-redux';
import { DraggableCore } from 'react-draggable';
import ReactDOM from 'react-dom';
import { filter } from 'lodash';
import DraggablePreview from '../../utils/DraggablePreview';

function getCoords(event, draggableData) {
  return {
    x: event.clientX,
    y: event.clientY,
  };
}

class Draggable extends Component {

  constructor(props) {
    super(props);

    this.state = {
      preview: {
        visible: false,
      },
    };
  }

  componentWillUnmount() {
    if (this.state.preview.preview) { this.state.preview.preview.cleanup(); }
  }

  onStart = (event, draggableData) => {
    const draggablePreview = new DraggablePreview(ReactDOM.findDOMNode(this).firstChild);

    const coords = getCoords(event, draggableData);

    this.setState({
      preview: {
        visible: true,
        preview: draggablePreview,
      }
    }, () => {
      this.state.preview.preview.position(coords);
    });
  }

  onDrag = (event, draggableData) => {
    if (!this.state.preview.preview) { return; }
    this.state.preview.preview.position(getCoords(event, draggableData));
  }

  onStop = (event, draggableData) => {
    if (this.state.preview.preview) { this.state.preview.preview.cleanup(); }

    this.setState({
      preview: {
        visible: false,
      }
    });

    const {
      x,
      y,
    } = getCoords(event, draggableData);

    const hits = filter(this.props.zones, (zone) => {
      return x > zone.x && x < zone.x + zone.width && y > zone.y && y < zone.y + zone.height
    });

    if (hits.length > 0) {
      this.props.onDropped(hits);
    }

  }

  render() {
    const opacity = this.state.preview.visible ? { opacity: 0 } : { opacity: 1 };

    return (
      <DraggableCore position={ { x: 0, y: 0 } } onStart={ this.onStart } onStop={ this.onStop } onDrag={ this.onDrag }>
        <div style={ opacity }>
          { this.props.children }
        </div>
      </DraggableCore>
    );
  }
}

function mapStateToProps(state) {
  return {
    zones: state.droppable.zones,
  }
}

export default connect(mapStateToProps)(Draggable);
