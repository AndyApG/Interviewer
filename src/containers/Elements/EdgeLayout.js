import React from 'react';
import PropTypes from 'prop-types';
import { find } from 'lodash';

const getCoords = (nodes, edge) => {
  console.log(edge);
  const from = find(nodes, ['id', edge.from]);
  const to = find(nodes, ['id', edge.to]);

  if (!from || !to) { return { error: true }; }

  return {
    from: from.closenessLayout,
    to: to.closenessLayout,
  };
};

const EdgeLayout = ({ nodes, edges }) => (
  <div className="edge-layout">
    <svg viewBox="0 0 1 1" xmlns="http://www.w3.org/2000/svg" preserveAspectRatio="none">
      <rect x="0" y="0" width="0.1" height="0.1" />
      { edges.map((edge, key) => {
        const { from, to, error } = getCoords(nodes, edge);
        if (error) { return null; }
        return <line key={key} x1={from.x} y1={from.y} x2={to.x} y2={to.y} />;
      }) }
    </svg>
  </div>
);

EdgeLayout.propTypes = {
  edges: PropTypes.array.isRequired,
  nodes: PropTypes.array.isRequired,
};

export default EdgeLayout;
