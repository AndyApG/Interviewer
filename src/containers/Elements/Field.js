/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';
import { Field as ReduxFormField } from 'redux-form';
import { map, toPairs } from 'lodash';
import {
  TextInput as Alphanumeric,
  RadioGroup,
  ToggleGroup as SwitchGroup,
} from 'network-canvas-ui';

import validations from '../../utils/Validations';

import { withOptionsFromSelector } from '../../behaviours';

/**
  * Returns the named field compontent, if no matching one is found
  or else it just returns a text input
  * @param {object} field The properties handed down from the protocol form
  */

export const renderInput = (field) => {
  const {
    type,
    input,
    meta,
    label,
    options,
    optionsSelector,
  } = field;
  let InputComponent = Alphanumeric;

  let inputProps = {
    name: input.name,
    value: input.value,
    errorText: meta.invalid && meta.dirty && meta.error,
    label,
  };

  if (type === 'Numeric') {
    inputProps = {
      ...inputProps,
      isNumericOnly: true,
    };
  }

  if (type === 'RadioGroup') {
    InputComponent = RadioGroup;
    inputProps = {
      ...inputProps,
      options,
      onRadioClick: input.onChange,
    };
  }

  if (type === 'CheckboxGroup') {
    InputComponent = SwitchGroup;
    inputProps = {
      ...inputProps,
      toggleComponent: 'checkbox',
      options,
      onOptionClick: (e, checked, optionVal) => input.onChange({
        ...input.value,
        [optionVal]: checked,
      }),
    };
  }

  if (type === 'ToggleGroup') {
    const { colors } = field;
    InputComponent = SwitchGroup;
    inputProps = {
      ...inputProps,
      toggleComponent: 'context',
      options,
      colors,
      onOptionClick: (e, checked, optionVal) => input.onChange({
        ...input.value,
        [optionVal]: checked,
      }),
    };
  }

  if (optionsSelector) {
    InputComponent = withOptionsFromSelector(InputComponent, optionsSelector);
  }

  return (
    <InputComponent
      {...inputProps}
      {...input}
    />
  );
};

/**
* Returns the named validation function, if no matching one is found it returns a validation
* which will always fail.
* @param {string} validation The name of the validation function to return.
  */
const getValidation = validation =>
  map(
    toPairs(validation),
    ([type, options]) => (
      Object.hasOwnProperty.call(validations, type) ? validations[type](options) : () => (`Validation "${type}" not found`)
    ),
  );

/**
  * Renders a redux-form field in the style of our app.
  * @param {string} label Presentational label
  * @param {string} name Property name
  * @param {string} type Field component type
  * @param {string} placeholder Presentational placeholder text
  * @param {object} validation Validation methods
  */
const Field = ({ label, name, type, validation, optionsSelector, ...rest }) => {
  const validate = getValidation(validation);

  return (
    <ReduxFormField
      name={name}
      label={label}
      component={renderInput}
      validate={validate}
      optionsSelector={optionsSelector}
      type={type}
      {...rest}
    />
  );
};

Field.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  optionsSelector: PropTypes.func,
  validation: PropTypes.object,
};

Field.defaultProps = {
  validation: {},
  optionsSelector: null,
};

export default Field;
