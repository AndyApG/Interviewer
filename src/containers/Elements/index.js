export { default as Panels } from './Panels';
export { default as Panel } from './Panel';
export { default as PromptSwiper } from './PromptSwiper';
