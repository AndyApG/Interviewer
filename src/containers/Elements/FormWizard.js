import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'network-canvas-ui';
import { Pips } from '../../components/Elements';
import { Form } from '.';

class FormWizard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      fieldIndex: 0,
    };
  }

  onSubmit = (formData, dispatch, form) => {
    if (this.shouldShowNextField()) {
      this.nextField();
      return;
    }
    this.props.onSubmit(formData, dispatch, form);
  }

  nextField = () => {
    const count = this.props.fields.length;
    this.setState({
      fieldIndex: (this.state.fieldIndex + 1 + count) % count,
    });
  };

  previousField = () => {
    const count = this.props.fields.length;
    this.setState({
      fieldIndex: (this.state.fieldIndex - 1 + count) % count,
    });
  };

  shouldShowNextField = () => {
    const showingLastField = this.state.fieldIndex === this.props.fields.length - 1;
    return !showingLastField;
  }

  render() {
    const { fields, onSubmit, ...rest } = this.props;

    const extraProps = {};
    extraProps.fields = [fields[this.state.fieldIndex]];
    extraProps.onSubmit = this.onSubmit;
    if (this.shouldShowNextField()) {
      extraProps.submitComponent = (
        <button className="form__next-button" aria-label="Submit">
          <Icon name="form-arrow-right" />
        </button>
      );
    }

    return (
      <div>
        <div className="modal__pips">
          <Pips count={fields.length} currentIndex={this.state.fieldIndex} />
        </div>
        <div className="modal--mobile__previous">
          {this.state.fieldIndex !== 0 && <Icon name="form-arrow-left" onClick={this.previousField} />}
        </div>
        <Form
          {...rest}
          {...extraProps}
        />
      </div>
    );
  }
}

FormWizard.propTypes = {
  fields: PropTypes.array.isRequired,
  onSubmit: PropTypes.func.isRequired,
  submitComponent: PropTypes.object.isRequired,
};

export default FormWizard;
