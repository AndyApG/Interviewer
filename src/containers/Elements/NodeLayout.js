import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators, compose } from 'redux';
import { connect } from 'react-redux';
import { first, isMatch } from 'lodash';
import { Node } from 'network-canvas-ui';
import { draggable, withBounds, selectable } from '../../behaviours';
import { DropZone } from '../../components/Elements';
import { activePrompt } from '../../selectors/session';
import { actionCreators as networkActions } from '../../ducks/modules/network';

const label = node => node.nickname;

const draggableType = 'POSITIONED_NODE';

const EnhancedNode = draggable(selectable(Node));

export class NodeLayout extends Component {
  constructor(props) {
    super(props);

    this.state = {
      connectFrom: null,
    };
  }

  onSelectNode = (node) => {
    if (!this.props.prompt.selectAction) { return; }

    switch (this.props.prompt.selectAction) {
      case 'EDGE':
        this.connectNode(node); break;
      case 'ATTRIBUTES':
        this.toggleNodeAttributes(node); break;
      default:
    }
  };

  updateNodeLayout = (hits, coords, node) => {
    const { prompt, updateNode } = this.props;
    const hit = first(hits);
    const relativeCoords = {
      x: (coords.x - hit.x) / hit.width,
      y: (coords.y - hit.y) / hit.height,
    };

    updateNode({ ...node, [prompt.layout]: relativeCoords });
  }

  connectNode(node) {
    const nodeId = node.id;
    const edgeType = this.props.prompt.edgeType;
    const connectFrom = this.state.connectFrom;


    if (!connectFrom) {
      this.setState({ connectFrom: nodeId });
      return;
    }

    if (connectFrom !== nodeId) {
      this.props.toggleEdge({
        from: connectFrom,
        to: nodeId,
        type: edgeType,
      });
    }

    this.setState({ connectFrom: null });
  }

  toggleNodeAttributes(node) {
    this.props.toggleNodeAttributes(node, this.props.prompt.nodeAttributes);
  }

  isSelected(node) {
    switch (this.props.prompt.selectAction) {
      case 'EDGE':
        return (node.id === this.state.connectFrom);
      case 'ATTRIBUTES':
        return isMatch(node, this.props.prompt.nodeAttributes);
      default:
        return false;
    }
  }

  render() {
    const { prompt, nodes, width, height } = this.props;

    const canDrag = Object.prototype.hasOwnProperty.call(prompt, 'canDrag') ? prompt.canDrag : true;
    const canSelect = Object.prototype.hasOwnProperty.call(prompt, 'canSelect') ? prompt.canSelect : true;

    return (
      <DropZone droppableName="NODE_LAYOUT" acceptsDraggableType={draggableType}>
        <div className="node-layout">
          { nodes.map((node, key) => {
            if (!Object.prototype.hasOwnProperty.call(node, prompt.layout)) { return null; }

            const x = node[prompt.layout].x * width;
            const y = node[prompt.layout].y * height;

            return (
              <div key={key} className="node-layout__node" style={{ left: `${x}px`, top: `${y}px` }}>
                <EnhancedNode
                  label={label(node)}
                  draggableType={draggableType}
                  onDropped={(hits, coords) => this.updateNodeLayout(hits, coords, node)}
                  onDrag={(hits, coords) => this.updateNodeLayout(hits, coords, node)}
                  onSelected={() => this.onSelectNode(node)}
                  selected={this.isSelected(node)}
                  canDrag={canDrag}
                  canSelect={canSelect}
                  animate={false}
                  {...node}
                />
              </div>
            );
          }) }
        </div>
      </DropZone>
    );
  }
}

NodeLayout.propTypes = {
  nodes: PropTypes.array,
  updateNode: PropTypes.func.isRequired,
  toggleEdge: PropTypes.func.isRequired,
  toggleNodeAttributes: PropTypes.func.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  prompt: PropTypes.object.isRequired,
};

NodeLayout.defaultProps = {
  nodes: [],
};

function mapStateToProps(state) {
  return {
    prompt: activePrompt(state),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    updateNode: bindActionCreators(networkActions.updateNode, dispatch),
    toggleNodeAttributes: bindActionCreators(networkActions.toggleNodeAttributes, dispatch),
    toggleEdge: bindActionCreators(networkActions.toggleEdge, dispatch),
  };
}

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withBounds,
)(NodeLayout);
