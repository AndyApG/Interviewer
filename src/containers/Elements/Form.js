import React, { Component } from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { reduxForm, getFormValues, getFormMeta } from 'redux-form';
import { Button } from 'network-canvas-ui';
import { autoInitialisedForm } from '../../behaviours';
import { Field } from '../../containers/Elements';
import { makeRehydrateFields } from '../../selectors/rehydrate';

/**
  * Renders a redux form that contains fields according to a `fields` config.
  *
  * @param {bool} addAnother tells whether or not there should be an "Add Another" button
  * @param {func} autoPopulate(fields, values, autofill) Enables prepopulation of fields
  * based on field value changes. Called on change with current field values and meta,
  * and a callback to allow the setting of otherfields
  * @param {func} continuousSubmit() handles continuous submission
  * @param {array} fields Contains an array of field definitions
  * see Field for detailed format of definitions:
  * {
  *   label: 'Name',
  *   name: 'name',
  *   type: 'Alphanumeric',
  *   placeholder: 'Name',
  *   validation: {
  *     required: true,
  *   }
  * }
  * @param {string} form The name of the form
  * @param {func} handleSubmit(data) Recieves data as jsonfrom a sucessful form submission
  * @param {component} next An optional component to trigger the next field in a wizard
  * @param {component} previous An optional component to trigger the previous field in a wizard
  *
  */
class Form extends Component {
  handleFieldBlur = () => {
    if (!this.props.autoPopulate) { return; }

    const {
      meta: {
        fields,
        values,
      },
      dirty,
      autofill,
    } = this.props;

    // if we don't check dirty state, this ends up firing and auto populating fields
    // when it shouldn't, like when closing the form
    if (dirty) {
      this.props.autoPopulate(fields, values, autofill);
    }
  };

  render() {
    const {
      addAnother,
      autoFocus,
      continuousSubmit,
      fields,
      handleSubmit,
      next,
      normalSubmit,
      previous,
    } = this.props;

    const addAnotherButton = addAnother && !next
      ? <Button color="white" onClick={continuousSubmit} aria-label="Submit and add another node">Submit and New</Button>
      : null;

    return (
      <form onSubmit={handleSubmit}>
        {previous}
        { fields.map((field, index) => {
          const isFirst = autoFocus && index === 0;
          return (
            <Field
              key={field.name}
              {...field}
              autoFocus={isFirst}
              onBlur={() => { this.handleFieldBlur(); }}
            />
          );
        }) }
        <div className="form__button-container">
          {addAnotherButton}
          {next ?
            <button className="form__next-button" onClick={normalSubmit} aria-label="Submit">{next}</button> :
            <Button onClick={normalSubmit} aria-label="Submit">Submit</Button>
          }
        </div>
      </form>
    );
  }
}

Form.propTypes = {
  addAnother: PropTypes.bool,
  autofill: PropTypes.func.isRequired,
  autoFocus: PropTypes.bool,
  autoPopulate: PropTypes.func,
  continuousSubmit: PropTypes.func,
  dirty: PropTypes.bool.isRequired,
  form: PropTypes.string.isRequired,
  fields: PropTypes.array.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  meta: PropTypes.object.isRequired,
  next: PropTypes.object,
  normalSubmit: PropTypes.func,
  previous: PropTypes.object,
};

Form.defaultProps = {
  addAnother: false,
  autoFocus: false,
  autoPopulate: null,
  continuousSubmit: null,
  next: null,
  normalSubmit: null,
  previous: null,
};

function makeMapStateToProps() {
  const rehydrateFields = makeRehydrateFields();

  return function mapStateToProps(state, props) {
    return {
      meta: {
        fields: getFormMeta(props.form)(state),
        values: getFormValues(props.form)(state),
      },
      fields: rehydrateFields(state, props),
    };
  };
}

export default compose(
  connect(makeMapStateToProps),
  autoInitialisedForm,
  reduxForm({
    destroyOnUnmount: false, // need this false to make it validate across wizard
    enableReinitialize: true,
    forceUnregisterOnUnmount: true,
    keepDirtyOnReinitialize: true,
  }),
)(Form);
