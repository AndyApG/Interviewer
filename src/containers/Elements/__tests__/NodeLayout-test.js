/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { NodeLayout } from '../../Elements/NodeLayout';

const mockProps = {
  nodes: [],
  updateNode: () => {},
  toggleEdge: () => {},
  toggleNodeAttributes: () => {},
  width: 123,
  height: 456,
  prompt: {},
};

describe('<NodeLayout />', () => {
  it('renders ok', () => {
    const component = shallow(<NodeLayout {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
