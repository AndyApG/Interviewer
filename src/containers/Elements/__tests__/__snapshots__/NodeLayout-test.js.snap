// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NodeLayout /> renders ok 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(Droppable)
    acceptsDraggableType="POSITIONED_NODE"
    droppableName="NODE_LAYOUT"
>
    <div
        className="node-layout"
    >
        <Connect(LayoutNode)
            animate={false}
            areaHeight={456}
            areaWidth={123}
            canDrag={false}
            canSelect={false}
            draggableType="POSITIONED_NODE"
            layoutVariable="foo"
            node={
                Object {
                    "bar": "buzz",
                    "foo": Object {
                      "x": 0,
                      "y": 0,
                    },
                    "uid": 123,
                  }
            }
            onDropped={[Function]}
            onSelected={[Function]}
            selected={false}
        />
    </div>
</Connect(Droppable)>,
  "nodes": Array [
    <Connect(Droppable)
      acceptsDraggableType="POSITIONED_NODE"
      droppableName="NODE_LAYOUT"
>
      <div
            className="node-layout"
      >
            <Connect(LayoutNode)
                  animate={false}
                  areaHeight={456}
                  areaWidth={123}
                  canDrag={false}
                  canSelect={false}
                  draggableType="POSITIONED_NODE"
                  layoutVariable="foo"
                  node={
                        Object {
                              "bar": "buzz",
                              "foo": Object {
                                "x": 0,
                                "y": 0,
                              },
                              "uid": 123,
                            }
                  }
                  onDropped={[Function]}
                  onSelected={[Function]}
                  selected={false}
            />
      </div>
</Connect(Droppable)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <NodeLayout
        allowPositioning={false}
        allowSelect={false}
        canCreateEdge={false}
        canHighlight={false}
        concentricCircles={0}
        createEdge="bar"
        displayEdges={Array []}
        height={456}
        highlightAttributes={Object {}}
        layoutVariable="foo"
        nodeBinSortOrder={Object {}}
        nodes={
                Array [
                        Object {
                          "bar": "buzz",
                          "foo": Object {
                            "x": 0,
                            "y": 0,
                          },
                          "uid": 123,
                        },
                      ]
        }
        selectMode=""
        skewedTowardCenter={false}
        toggleEdge={[Function]}
        toggleHighlight={[Function]}
        updateNode={[Function]}
        width={123}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": NodeLayout {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onDropNode": [Function],
        "onSelectNode": [Function],
        "props": Object {
          "allowPositioning": false,
          "allowSelect": false,
          "canCreateEdge": false,
          "canHighlight": false,
          "concentricCircles": 0,
          "createEdge": "bar",
          "displayEdges": Array [],
          "height": 456,
          "highlightAttributes": Object {},
          "layoutVariable": "foo",
          "nodeBinSortOrder": Object {},
          "nodes": Array [
            Object {
              "bar": "buzz",
              "foo": Object {
                "x": 0,
                "y": 0,
              },
              "uid": 123,
            },
          ],
          "selectMode": "",
          "skewedTowardCenter": false,
          "toggleEdge": [Function],
          "toggleHighlight": [Function],
          "updateNode": [Function],
          "width": 123,
        },
        "refs": Object {},
        "state": Object {
          "connectFrom": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(Droppable)
          acceptsDraggableType="POSITIONED_NODE"
          droppableName="NODE_LAYOUT"
>
          <div
                    className="node-layout"
          >
                    <Connect(LayoutNode)
                              animate={false}
                              areaHeight={456}
                              areaWidth={123}
                              canDrag={false}
                              canSelect={false}
                              draggableType="POSITIONED_NODE"
                              layoutVariable="foo"
                              node={
                                        Object {
                                                  "bar": "buzz",
                                                  "foo": Object {
                                                    "x": 0,
                                                    "y": 0,
                                                  },
                                                  "uid": 123,
                                                }
                              }
                              onDropped={[Function]}
                              onSelected={[Function]}
                              selected={false}
                    />
          </div>
</Connect(Droppable)>,
        "_debugID": 2,
        "_renderedOutput": <Connect(Droppable)
          acceptsDraggableType="POSITIONED_NODE"
          droppableName="NODE_LAYOUT"
>
          <div
                    className="node-layout"
          >
                    <Connect(LayoutNode)
                              animate={false}
                              areaHeight={456}
                              areaWidth={123}
                              canDrag={false}
                              canSelect={false}
                              draggableType="POSITIONED_NODE"
                              layoutVariable="foo"
                              node={
                                        Object {
                                                  "bar": "buzz",
                                                  "foo": Object {
                                                    "x": 0,
                                                    "y": 0,
                                                  },
                                                  "uid": 123,
                                                }
                              }
                              onDropped={[Function]}
                              onSelected={[Function]}
                              selected={false}
                    />
          </div>
</Connect(Droppable)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <NodeLayout
    allowPositioning={false}
    allowSelect={false}
    canCreateEdge={false}
    canHighlight={false}
    concentricCircles={0}
    createEdge="bar"
    displayEdges={Array []}
    height={456}
    highlightAttributes={Object {}}
    layoutVariable="foo"
    nodeBinSortOrder={Object {}}
    nodes={
        Array [
            Object {
              "bar": "buzz",
              "foo": Object {
                "x": 0,
                "y": 0,
              },
              "uid": 123,
            },
          ]
    }
    selectMode=""
    skewedTowardCenter={false}
    toggleEdge={[Function]}
    toggleHighlight={[Function]}
    updateNode={[Function]}
    width={123}
/>,
}
`;
